packages/svelte/src/remix-router-svelte.ts

```diff
@@ -7,12 +7,104 @@ import {
   type Navigation,
   type Router,
   type Location,
+  type HydrationState,
+  createRouter,
+  createMemoryHistory,
+  type AgnosticRouteObject,
+  type AgnosticRouteMatch,
+  createBrowserHistory,
+  createHashHistory,
 } from "@remix-run/router";
-import { onDestroy } from "svelte";
+import { onDestroy, type SvelteComponent } from "svelte";
 import { derived, get, writable, type Readable } from "svelte/store";
 import { getRouteContext, getRouterContext } from "./contexts";
 import { getFormSubmissionInfo, type SubmitOptions } from "./dom";
 
+// Create svelte-specific types from the agnostic types in @remix-run/router to
+// export from remix-router-svelte
+export interface RouteObject extends AgnosticRouteObject {
+  children?: RouteObject[];
+  element?: SvelteComponent | null;
+  errorElement?: SvelteComponent | null;
+}
```

Comment:
In one of the recent releases we got the last react-specific typings out of `@remix-run/router`, so each package now extends `AgnosticRouteObject` with it's own framework-specific typings.

Reply:
There's a weird nuance with the `svelte` types where you actually need to use `typeof SvelteComponent` for this to work as expected. You'll see the TypeScript errors if you open `reference-app/App.svelte` and switch the `RouteObject` import to use `remix-router-svelte`.